assets/js/app.js

```diff
@@ -0,0 +1,73 @@
+String.prototype.shuffle = function () {
+  var a = this.split(""),
+    n = a.length;
+
+  for (var i = n - 1; i > 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
+    var tmp = a[i];
+    a[i] = a[j];
+    a[j] = tmp;
+  }
+  return a.join("");
+}
+
+function generatePassword() {
+
+  let passwordLength = parseInt(document.querySelector("#inputPasswordLength").value);
+  if (!(passwordLength == NaN) && (passwordLength >= 8) && (passwordLength <= 128)) {
+
+    let charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    let charsetLower = "abcdefghijklmnopqrstuvwxyz"
+    let charsetUpper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    let charsetNumber = "0123456789"
+    let charsetSpecial = "!§$%&/()=?{}"
+    let genPassword = ""
+    let genPasswordLen = passwordLength
+
+    let hasNumber = document.querySelector("#numberCheck").checked
+    let hasUpperCase = document.querySelector("#upperCaseCheck").checked
+    let hasLowerCase = document.querySelector("#lowerCaseCheck").checked
+    let hasSpecialCharacter = document.querySelector("#specialCharacterCheck").checked
+
+    if (hasNumber) {
```

Comment:
From here to line 51 then if statement will only run 1 char from each selected option. to change this you can add the checked char to charset and charset would remain empty